libraries/botbuilder-dialogs-adaptive-testing/src/httpRequestMocks/httpRequestSequenceMock.ts

```diff
@@ -0,0 +1,76 @@
+/**
+ * @module botbuilder-dialogs-adaptive-testing
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { HttpMethod } from 'botbuilder-dialogs-adaptive';
+import * as nock from 'nock';
+import * as parse from 'url-parse';
+import { HttpRequestMock } from './httpRequestMock';
+import { HttpResponseMock } from './httpResponseMock';
+import { SequenceResponseManager } from './sequenceResponseManager';
+
+export enum BodyMatchType {
+    Exact = 'Exact',
+    Partial = 'Partial',
+}
+
+export interface HttpRequestSequenceMockConfiguration {
+    method?: string | HttpMethod;
+    url?: string;
+    matchType?: string | BodyMatchType;
+    body?: string;
+    responses?: HttpResponseMock[];
+}
+
+export class HttpRequestSequenceMock extends HttpRequestMock implements HttpRequestSequenceMockConfiguration {
+    public static $kind = 'Microsoft.Test.HttpRequestSequenceMock';
+
+    public method: HttpMethod;
+    public url: string;
+    public matchType: BodyMatchType = BodyMatchType.Partial;
+    public body = '';
+    public responses: HttpResponseMock[] = [];
+
+    public setup(): void {
+        if (this.body && (this.method === HttpMethod.DELETE || this.method === HttpMethod.GET)) {
+            throw new Error("GET and DELETE don't support matching body!");
+        }
+        const response = new SequenceResponseManager(this.responses);
+        const url = parse(this.url);
+        const path = (url.pathname || '/') + url.query;
```

Comment:
`??` instead of `||`

Reply:
Also, do we expect `url.query` to include the `?`

Reply:
the `pathname` could be an empty string, so I have to use `||` to check if it is empty.

Reply:
The `?` should be included in `url.query`:
```
> parse('http://localhost:4443/api?abc=123')
{ slashes: true,
  protocol: 'http:',
  hash: '',
  query: '?abc=123',
  pathname: '/api',
  auth: '',
  host: 'localhost:4443',
  port: '4443',
  hostname: 'localhost',
  password: '',
  username: '',
  origin: 'http://localhost:4443',
  href: 'http://localhost:4443/api?abc=123' }
```